packages/grid/x-data-grid-pro/src/hooks/features/columnResize/useGridColumnResize.tsx

```diff
@@ -136,7 +136,7 @@ export const useGridColumnResize = (
 
     colDefRef.current!.computedWidth = newWidth;
     colDefRef.current!.width = newWidth;
-    colDefRef.current!.flex = undefined;
+    colDefRef.current!.flex = 0;
```

Comment:
This change doesn't seem necessary to fix #4310 but makes sense to set `0`.

Reply:
When I did not do this one, I had the flex coming back on re-import.
Maybe it only occurs in some scenario.

As `0` make sense, I propose to keep it.